.github/workflows/build-pull-request.yml

```diff
@@ -26,6 +26,12 @@ jobs:
     with:
       cache-read-only: true
 
+  test-native:
+    uses: ./.github/workflows/reusable-native-tests.yml
+    with:
+      skip-native-tests: ${{ !contains(github.event.pull_request.labels.*.name, 'test native') }}
+      cache-read-only: true
```

Comment:
I suspect that `cache-read-only` doesn't do anything here as `reusable-native-tests.yml` does not declare it as an input. Another thing to consider is that the caching only works when you are using `gradle/gradle-build-action` or `actions/cache`